uavcan/node/62805.Status.1.0.uavcan

```diff
@@ -0,0 +1,47 @@
+#
+# Abstract node status information.
+#
+# All UAVCAN nodes are required to publish this message periodically.
+# This is the only data type that must be supported by all nodes.
+#
+# The default subject ID 62805 is 1111010101010101 in binary. The alternating bit pattern at the end
+# helps transceiver synchronization (e.g., on CAN-based networks) and on some transports permits
+# automatic bit rate detection (e.g., CAN bus).
+#
+
+# Publication period must not exceed this limit.
+# The period should not change as long as the node is running.
+uint16 MAX_PUBLICATION_PERIOD = 1   # [second]
+
+# If a node fails to publish this message in this amount of time, it should be considered offline.
+uint16 OFFLINE_TIMEOUT = 3          # [second]
+
+# The uptime seconds counter should never overflow. The counter will reach the upper limit in ~136 years,
+# upon which time it should stay at 0xFFFFFFFF until the node is restarted.
+#
+# Other nodes may detect that a remote node has restarted when this value jumps backwards.
+uint32 uptime                       # [second]
+
+# Abstract node health.
+uint2 HEALTH_OK         = 0         # Nominal. Color: none or green.
+uint2 HEALTH_ADVISORY   = 1         # Minor parameter deviation or minor issue. Color: blue/cyan.
```

Comment:
The HEALTH_ADVISORY doesn't really make any intuitive sense. Sounds like a [committee](https://health.nt.gov.au/health-governance/health-advisory-committees)

Reply:
This is derived from https://www.faa.gov/regulations_policies/rulemaking/committees/documents/media/TAEas-wcal-04232002.pdf

Reply:
When I searched through the document I found a lot of references to "advisory light" which makes sense. I found little references to "node" or "health". Perhaps it's the "health" name that seems a bit off?

----
The first thing that comes to mind is the following

```python
# State used for indicating node status.
# Used in lights, panels, etc.
uint2 INDICATION_OK               = 0         # Nominal. Color: none or green.
uint2 INDICATION_ADVISORY   = 1         # Minor parameter deviation or minor issue. Color: blue/cyan.
uint2 INDICATION_CAUTION    = 2         # Major problem or failure. Color: amber/yellow.
uint2 INDICATION_WARNING   = 3         # Fault, loss of function. Color: red.
truncated uint2 indication
```

---

A different alternative would be to call the message `Heartbeat`, which is a pretty good description of what it is. And call the field `status`

```python
# The node status. Should be indicated with recommended color in lights, panels, etc.
uint2 STATUS_OK               = 0         # Nominal. Color: none or green.
uint2 STATUS_ADVISORY   = 1         # Minor parameter deviation or minor issue. Color: blue/cyan.
uint2 STATUS_CAUTION    = 2         # Major problem or failure. Color: amber/yellow.
uint2 STATUS_WARNING   = 3         # Fault, loss of function. Color: red.
truncated uint2 status
```

Reply:
That makes sense. I have renamed the message to `Heartbeat` and renamed the field from `health` to `health_status`.